File: api/src/main/java/net/blueshell/api/dto/ContributionPeriodDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

@Data
@EqualsAndHashCode(callSuper = false)
@Schema(name = "ContributionPeriod")
public class ContributionPeriodDTO extends BaseDTO {
    @Nullable
    private Long id;

    @NotNull
    private LocalDate startDate;

    @NotNull
    private LocalDate endDate;

    @NotNull
    private double halfYearFee;

    @NotNull
    private double fullYearFee;

    @NotNull
    private double alumniFee;

    private Long listId;
}

File: api/src/main/kotlin/net/blueshell/api/dto/contribution/ContributionPeriodDTO.kt
package net.blueshell.api.dto.contribution

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import net.blueshell.api.dto.base.AuditedAutoIdDTO
import java.time.LocalDate

@Schema(name = "ContributionPeriod")
data class ContributionPeriodDTO(
    @field:NotNull
    var startDate: LocalDate,

    @field:NotNull
    var endDate: LocalDate,

    @field:NotNull
    var halfYearFee: Double = 0.0,

    @field:NotNull
    var fullYearFee: Double = 0.0,

    @field:NotNull
    var alumniFee: Double = 0.0,

    var listId: Long? = null
) : AuditedAutoIdDTO()
